### PR TITLE
cmake v2 plugin: default to -DCMAKE_BUILD_TYPE=Release

### PR DESCRIPTION
**Pull request by sergiusens**
Expose an environment variable to easily override and ignore
completely if part of cmake-parameters.

Additionally, remove the incorrect "minitems" in the schema.

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: snapcraft/plugins/v2/cmake.py [hunk 1]
```diff
@@ -28,6 +28,17 @@
     - cmake-parameters:
       (list of strings)
       parameters to pass to the build using the common cmake semantics.
+
+This plugin also interprets these specific build-environment entries:
+
+    - SNAPCRAFT_CMAKE_INSTALL_PREFIX
```
Comment (cjp256) [inline snapcraft/plugins/v2/cmake.py:34]: I feel like the lines are a bit blurred between schema and the variables...

Just wondering - have you considered to instead set the default list of `cmake-parameters: ["-DCMAKE_INSTALL_PREFIX=/", "-DCMAKE_BUILD_TYPE=Release"] if unset?  It provides a path for the user to override it, with the documentation reinforcing their consideration if overriding?  Seems to me to cover the 90% case without complexity.

Comment (sergiusens) [reply snapcraft/plugins/v2/cmake.py:34]: I have, the problem with that is exactly the overriding, most of the user errors and complaints we had from field is the fact that either things cannot be replaced or things cannot be added. This solution seems to cater for both those counter complaints.

Comment (sergiusens) [reply snapcraft/plugins/v2/cmake.py:34]: And yes, the lines are a bit blurred, this was my main concern with exposing both environment variables and "options" in the first place.

File: snapcraft/plugins/v2/cmake.py [hunk 2]
```diff
@@ -69,7 +70,12 @@ def _get_cmake_configure_command(self) -> str:
             c.startswith("-DCMAKE_INSTALL_PREFIX=")
             for c in self.options.cmake_parameters
         ):
-            cmd.append('-DCMAKE_INSTALL_PREFIX="${SNAPCRAFT_CMAKE_INSTALL_PREFIX}"')
+            cmd.append("-DCMAKE_INSTALL_PREFIX=/")
+
+        if not any(
+            c.startswith("-DCMAKE_BUILD_TYPE=") for c in self.options.cmake_parameters
+        ):
+            cmd.append("-DCMAKE_BUILD_TYPE=Release")
```
Comment (cjp256) [inline snapcraft/plugins/v2/cmake.py:78]: Last Q, do we really need to even specify this at all?  Perhaps just leave it up to the project?

Comment (sergiusens) [reply snapcraft/plugins/v2/cmake.py:78]: ok, will remove, but this is the point of no return.

**Pull request closed without merging**